### PR TITLE
[MBL-16881][Student] Offline People Details

### PR DESCRIPTION
**Pull request by domonkosadam**
Test plan:

refs: MBL-16881
affects: Student
release note: none

## Checklist

- [x] Tested in dark mode
- [x] Tested in light mode
- [x] A11y checked

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: apps/student/src/main/java/com/instructure/student/features/people/details/PeopleDetailsDataSource.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ */
+
+package com.instructure.student.features.people.details
+
+import com.instructure.canvasapi2.models.CanvasContext
+import com.instructure.canvasapi2.models.User
+import com.instructure.canvasapi2.utils.DataResult
+
+interface PeopleDetailsDataSource {
+    suspend fun loadUser(canvasContext: CanvasContext, userId: Long, forceNetwork: Boolean = false): DataResult<User>
```
Comment (hermannakos) [inline apps/student/src/main/java/com/instructure/student/features/people/details/PeopleDetailsDataSource.kt:25]: You could return the `User` object here

File: apps/student/src/main/java/com/instructure/student/features/people/details/PeopleDetailsFragment.kt
Comment (hermannakos) [inline apps/student/src/main/java/com/instructure/student/features/people/details/PeopleDetailsFragment.kt:1]: You should move the permission api call to the repository as well.

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

File: apps/student/src/main/java/com/instructure/student/features/people/details/PeopleDetailsNetworkDataSource.kt [hunk 2]
```diff
@@ -19,13 +19,29 @@ package com.instructure.student.features.people.details
...
+    override suspend fun loadUser(canvasContext: CanvasContext, userId: Long, forceNetwork: Boolean): User {
         val restParams = RestParams(isForceReadFromNetwork = forceNetwork)
-        return userApi.getUserForContextId(canvasContext.apiContext(), canvasContext.id, userId, restParams)
+        return userApi.getUserForContextId(canvasContext.apiContext(), canvasContext.id, userId, restParams).dataOrThrow
+    }
+
+    override suspend fun loadMessagePermission(canvasContext: CanvasContext, user: User?): Boolean {
+        val id = canvasContext.id
+        return when {
+            canvasContext.isGroup -> GroupManager.getPermissionsAsync(id).awaitOrThrow().send_messages
```
Comment (tamaskozmer) [inline apps/student/src/main/java/com/instructure/student/features/people/details/PeopleDetailsNetworkDataSource.kt]: We are not meant to use these legacy manager classes in the new architecture. Please change it to the API instances and inject them in the constructor.

File: apps/student/src/main/java/com/instructure/student/features/people/details/PeopleDetailsFragment.kt [hunk 3]
```diff
@@ -97,7 +92,7 @@ class PeopleDetailsFragment : ParentFragment(), Bookmarkable {
             RouteMatcher.route(requireContext(), route)
         }
         when {
-            canvasContext.isCourse && user == null -> fetchUser()
+            canvasContext.isCourse -> fetchUser()
```
Comment (tamaskozmer) [inline apps/student/src/main/java/com/instructure/student/features/people/details/PeopleDetailsFragment.kt]: I think this was okay the way it was before. If we come from a group and already have a user we don't need to fetch it again.

**Review by tamaskozmer — CHANGES_REQUESTED** [review]

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by tamaskozmer — APPROVED** [review]

**Comment by tamaskozmer** [comment]
QA + 1

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Pull request merged**